dotnet/src/Experimental/Agents/KernelAgent.cs

```diff
@@ -0,0 +1,36 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.SemanticKernel.Experimental.Agents;
+
+/// <summary>
+/// Represent a base class for agents.
+/// </summary>
+public abstract class KernelAgent
+{
+    /// <summary>
+    /// The <see cref="Kernel"/> containing services, plugins, and other state for use throughout the operation.
+    /// </summary>
+    public Kernel Kernel { get; internal set; }
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="KernelAgent"/> class.
+    /// </summary>
+    /// <param name="kernel">The <see cref="Kernel"/> containing services, plugins, and other state for use by the agent.</param>
+    protected KernelAgent(Kernel kernel)
+    {
+        Verify.NotNull(kernel, nameof(kernel));
+        this.Kernel = kernel;
+    }
+
+    /// <summary>
+    /// Invokes the agent to process the given messages and generate a response.
+    /// </summary>
+    /// <param name="messages">A list of the messages for the agent to process.</param>
+    /// <param name="cancellationToken">An optional <see cref="CancellationToken"/> to cancel the operation.</param>
+    /// <returns>List of messages representing the agent's response.</returns>
+    public abstract Task<IReadOnlyList<AgentMessage>> InvokeAsync(IReadOnlyList<AgentMessage> messages, CancellationToken cancellationToken = default);
```

Comment:
We need to verify that the scenarios are supported. So far, I've identified three ways it can be done:  
1. If two or more assistant agents need to chat in the same thread, they can be configured, at creation time, to use the same thread. This way, all the messages passed to the InvokeAsync method of agent A or agent B will be added to the same shared thread. If no thread is configured for an agent, then every InvokeAsync on that agent will create a thread, add messages to it, execute run (OpenAI Assistants API run), get results, and remove the thread.  
2. We can add a special KernelContent type, let's say - AssistantThreadContent with only one property, ThreadId/Id, and agents can use it to pass information about the thread between each other.  
3. Each AgentMessage has the InternalMessage property, which will have the OpenAI Assistants API message with the ThreadId property. This property can be used by another assistant agent to identify a thread to work with.  
   
I like the first approach because it gives the consumer code more control over agent configuration and thread ownership and lifetime management. In contrast, with the other two approaches, it's not clear who owns the thread, who should create, and remove it.